### PR TITLE
Add nanopore reads preprocessing with Porechop

### LINKED ISSUE #21: Adapter and quality trimming with porechop is performed in a `longread_preprocessing.nf` subworkflow

### PR DESCRIPTION
**Pull request by ljmesi**
Resolves #21, resolves #29

This pull request includes addition of Porechop module inside of a long reads preprocessing subworkflow. The subworkflow is integrated into `taxprofiler.nf`. 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker` --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ On release, automated continuous integration tests run the pipeline on a full-si
    - Low complexity filtering
    - Host read removal
    - Run merging
+   - Adapter and quality trimming of Nanopore reads
```
Comment (jfy133) [inline README.md:37]: I don't think this is necessary as a separate point. You can combine it with the first subpoint (i.e., `- Adapter clipping and merging (short, and nanopore reads)

File: workflows/taxprofiler.nf [hunk 2]
```diff
@@ -40,7 +40,7 @@ include { INPUT_CHECK         } from '../subworkflows/local/input_check'
 
 include { DB_CHECK            } from '../subworkflows/local/db_check'
 include { FASTQ_PREPROCESSING } from '../subworkflows/local/preprocessing'
```
Comment (jfy133) [inline workflows/taxprofiler.nf:42]: Could you rename/update FASTQ_PREPROCESSING and the file itself to SHORTREAD_PREPROCESSING for consistency?

**Review by jfy133 — APPROVED** [review]
In principle looks good. Could you update/rename the short-read preprocessing subworkflows so they are clearly distinguished?

File: README.md (hunk 1 above)
Comment (ljmesi) [reply README.md]: Yes, I fixed now the README.md and updated the preprocessing subworkflow name.

File: workflows/taxprofiler.nf (hunk 2 above)
Comment (ljmesi) [reply workflows/taxprofiler.nf]: Yes, they are updated now.

File: workflows/taxprofiler.nf [hunk 3]
```diff
@@ -101,13 +101,24 @@ workflow TAXPROFILER {
     // PERFORM PREPROCESSING
     //
     if ( params.fastp_clip_merge ) {
```
Comment (jfy133) [inline workflows/taxprofiler.nf:103]: I will syncronise these parameters later :+1:

File: workflows/taxprofiler.nf [hunk 4]
```diff
@@ -185,14 +197,13 @@ workflow TAXPROFILER {
     workflow_summary    = WorkflowTaxprofiler.paramsSummaryMultiqc(workflow, summary_params)
     ch_workflow_summary = Channel.value(workflow_summary)
 
-    ch_multiqc_files = Channel.empty()
     ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))
```
Comment (jfy133) [inline workflows/taxprofiler.nf:200]:
```suggestion
    ch_multiqc_files = Channel.empty()
    ch_multiqc_files = ch_multiqc_files.mix(Channel.from(ch_multiqc_config))
```

**Review by jfy133 — APPROVED** [review]

**Pull request merged**